code/frameworks/nextjs/src/images/decorator.tsx

```diff
@@ -0,0 +1,18 @@
+import * as React from 'react';
+import type { Addon_StoryContext } from '@storybook/types';
+import { ImageContext } from './context';
+
+export const ImageDecorator = (
+  Story: React.FC,
+  { parameters }: Addon_StoryContext
+): React.ReactNode => {
+  if (!parameters.nextjs?.image) {
+    return <Story />;
+  }
+
+  return (
+    <ImageContext.Provider value={parameters.nextjs.image}>
+      <Story />
+    </ImageContext.Provider>
+  );
```

Comment:
I tried other methods such as using environment variables or traversing the React element tree and adding the loading property, but in the end using Context seemed to be the most straightforward approach.